### DISCUSSION TITLE
Dropdown icon X not appearing

### DISCUSSION THREAD

**Discussion by batata004** [fomantic/Fomantic-UI#383] [Modules (Modal, Dropdown, Popup, Calendar, Toast, ...)] [answered]
Hi,

I used "gulp build" and removed the "icon" component. I am importing my icons directly from Font Awesome. The problem is that any `dropdown` element on my page with `allowAdditions:true` usually should appear with X at the right side so the user can remove the addition. However if I dont import the "icon" component no X is shown. See image below where in the addition "Branco" the X does not appear.

![image](https://user-images.githubusercontent.com/8988923/51046660-d050ea00-15ad-11e9-86ea-8eb6301d1cdd.png)

However if I add manually at the Chrome Dev Tools the class "icon-cancel" it works fine. Check image below:

![image](https://user-images.githubusercontent.com/8988923/51046706-f2e30300-15ad-11e9-9db7-152dd71d758d.png)

I checked the documentation of dropdown and I see there is an option "selector" but I tried using it and nothing seemed to work:

`$("select[name='cores\[\]']").dropdown({allowAdditions:true,,selector:{remove:".icon-cancel"}});`

Any idea how I could accomplish that?

**Comment by lubber-de** [comment, marked as answer]
To make this work you currently also have to provide a custom template, which is rendering the labels and its integrated delete icon.
I have to admit, we should change the default template to make use of a classname setting instead of having this hardcoded. i think i'll prepare a PR to fix this.

So for now, please try
```javascript
$("select[name='cores\[\]']").dropdown({
	allowAdditions:true,
	selector:{
		remove: '> .label > .icon-cancel.icon',		
	},
	templates: {
		label: function(value, text, preserveHTML) {
			var
				escape = $.fn.dropdown.settings.templates.escape;
			return escape(text,preserveHTML) + '<i class="icon-cancel icon"></i>';
		}
	}
});
```

**Comment by batata004** [comment]
@lubber-de  I must say your suggestion worked like a charm! I cant believe how you pulled that. Even the documentation does not say anything about it, I just thought that changing the "selector" attribute would fix the problem but the template was indeed necessary.

Thanks a lot man, you are always saving the day :)
